connectors/source.py

```diff
@@ -113,7 +113,15 @@ def __init__(self, configuration):
         self.configuration.set_defaults(self.get_default_configuration())
 
     def __str__(self):
-        return f"Datasource `{self.__class__.__doc__}`"
+        return f"Datasource `{self.__class__.name()}`"
+
+    @staticmethod
```

Comment:
why a static method? should be a class variable

```
class MySource:
    name = "name"
    service_type = "service type"
```

Reply:
or if you want to force the class to define them, use class methods+property

```
>>> class BaseDataSource:
...     @classmethod
...     @property
...     def name(cls):
...         raise NotImplementedError
...
>>> class Source(BaseDataSource):
...     pass
...
>>>
>>> s = Source()
>>> s.name
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 5, in name
NotImplementedError
>>> class Source2(BaseDataSource):
...     @classmethod
...     @property
...     def name(cls):
...         return 'source 2'
...
>>> s = Source2()
>>> s.name
'source 2'
```

Reply:
From [here](https://docs.python.org/3.11/library/functions.html#classmethod), it's not supported to use @classmethod and @property together since 3.11:

>_Changed in version 3.11_: Class methods can no longer wrap other [descriptors](https://docs.python.org/3.11/glossary.html#term-descriptor) such as [property()](https://docs.python.org/3.11/library/functions.html#property).

We are on 3.10, but it's definitely not recommended. I will use class variables instead, which means it doesn't force the child class to override it. But I think that's OK because 1) ultimately it's us to review the PR 2) Neither `name` nor `service_type` here is used in any critical path.